middle_end/flambda2/types/meet_and_join_new.ml

```diff
@@ -1139,14 +1222,22 @@ and meet_row_like_for_blocks env
       TG.Row_like_for_blocks.t)
     ({ known_tags = known2; other_tags = other2; alloc_mode = alloc_mode2 } :
       TG.Row_like_for_blocks.t) : TG.Row_like_for_blocks.t meet_result =
+  let meet_shape env shape1 shape2 : _ meet_result =
+    if K.Block_shape.equal shape1 shape2 then Ok (Both_inputs, env) else Bottom
```

Comment:
Probably a comment in block_shape should mention that point saying that different shapes should really only occur when absolutely incompatible